### PR TITLE
Add a badge selector

### PR DESCRIPTION
**Pull request by hellcp-work**
This is a companion PR to #13169, it's based on it, but I want this PR to be just about the selector.
Screenshot from 2022-10-19 10-49-10

### CODE REVIEW

File: src/api/app/assets/javascripts/webui/badge.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  var badgeImageUrl = new URL(BASE_BADGE_URL); // jshint ignore:line
+  badgeImageUrl.searchParams.set('type', $('#badge-style-selector').val());
+  $('#badge-preview').attr("src", badgeImageUrl);
+  if ($('#badge-format-selector').val() === 'markdown') {
+    return '[![build result](' + badgeImageUrl + ')](' + BASE_PACKAGE_URL + ')'; // jshint ignore:line
+  } else if ($('#badge-format-selector').val() === 'html') {
+    return '<a href="' + BASE_PACKAGE_URL + '"><img src="' + badgeImageUrl + '" alt="build result"/></a>'; // jshint ignore:line
+  }
+  return badgeImageUrl;
+}
```
Comment (hennevogel) [inline src/api/app/assets/javascripts/webui/badge.js:12]: I think this is too much. People that build packages are capable of formating their links. I would display mardown and be done with it.

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: src/api/.jshintrc [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "Option" :   false, // Constructor for creating an HTMLOptionElement.
     "ui" :       false,
     "window" :   false,
+    "URL" :      false,
```
Comment (hennevogel) [inline src/api/.jshintrc:14]: You don't use this anymore

**Review by hennevogel — APPROVED** [review]

**Comment by hennevogel** [comment]
I like it

[build result](https://obs-reviewlab.opensuse.org/hellcp-work-badge-selector/package/show/home:Admin:hans/ctris)

**Comment by danidoni** [comment]
> > No idea why feature spec fails
> 
> Explore the [artifacts](https://app.circleci.com/pipelines/github/openSUSE/open-build-service/8616/workflows/6b14e3c0-5629-4a22-93bf-c3c212a8e1b2/jobs/100225/artifacts) to understand why.

My guess is that the ui takes longer than the timeout value to finish loading and replace the spinner with some content that the spec is waiting for...

**Comment by hennevogel** [comment]
> My guess is

Don't guess. Look at the artifacts. They tell you what's going on.

**Comment by hennevogel** [comment]
I guess we should fix #13320 first

**Review by danidoni — APPROVED** [review]

**Pull request merged**